### PR TITLE
MHV-49842 Authenticate to MHV FHIR using ICN

### PR DESCRIPTION
**Pull request by mmoyer-va**
## Summary

- We were previously using MHV Correlation ID to authenticate to MHV FHIR server. Switched to using ICN.
- Fixed some incorrect error invocations

### [MHV-49852](https://jira.devops.va.gov/browse/MHV-49852) - Authenticate to MHV FHIR using ICN instead of MHV Correlation ID ###

> Currently we are sending MHV Correlation ID to the security endpoint, when we are supposed to be sending the ICN.
> 
> Make this change in mhv_jwt_session_client.rb

## Testing done

- Updated existing spec tests to use ICN; re-created VCR cassettes
- Manually logged in to local instance and verified success

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
MHV Medical Records

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by mattwrightva — APPROVED** [review]
ok!

**Comment by patrickvinograd** [comment]
Is the jwt_session_client (so far) only used by the FHIR API? And is it a totally separate session cache than the SM/Rx APIs using the mhv_session_cache?

Wondering if it makes sense to use icn as the redis cache key here rather than MHV ID, just to avoid more layers of mapping confusion. 

(Ignore me if that's already the case)

File: lib/common/client/concerns/mhv_jwt_session_client.rb [hunk 1]
```diff
@@ -139,10 +140,10 @@ def auth_body
           {
             'appId' => '103',
             'appToken' => config.app_token,
-            'subject' => session.user_id.to_s,
+            'subject' => session.icn,
```
Comment (rmtolmach) [inline lib/common/client/concerns/mhv_jwt_session_client.rb:143]: `to_s` no longer needed here?

Comment (mmoyer-va) [reply lib/common/client/concerns/mhv_jwt_session_client.rb:143]: @rmtolmach that's correct. user_id is a numeric value while icn is a string that contains alphanumeric characters.

**Comment by mmoyer-va** [comment]
> Is the jwt_session_client (so far) only used by the FHIR API? And is it a totally separate session cache than the SM/Rx APIs using the mhv_session_cache?
> 
> Wondering if it makes sense to use icn as the redis cache key here rather than MHV ID, just to avoid more layers of mapping confusion.
> 
> (Ignore me if that's already the case)

Had a discussion about this w/ @patrickvinograd in Slack. We agreed that it probably makes the most sense to use ICN as the redis key. I will implement this in a future PR.

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**